ze_graph_ext.h

```diff
@@ -769,6 +771,58 @@ typedef struct _ze_graph_dditable_ext_1_6_t
 
 } ze_graph_dditable_ext_1_6_t;
 
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Extension version 1.7
+
+///////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+typedef ze_result_t (ZE_APICALL *ze_pfnGraphLoad_ext_t)(
+    ze_graph_handle_t hGraph    ///< [in] handle of the graph object
+    );
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Table of Graph functions pointers
+typedef struct _ze_graph_dditable_ext_1_7_t
+{
+    // version 1.0
+    ze_pfnGraphCreate_ext_t                     pfnCreate;
+    ze_pfnGraphDestroy_ext_t                    pfnDestroy;
+    ze_pfnGraphGetProperties_ext_t              pfnGetProperties;
+    ze_pfnGraphGetArgumentProperties_ext_t      pfnGetArgumentProperties;
+    ze_pfnGraphSetArgumentValue_ext_t           pfnSetArgumentValue;
+    ze_pfnAppendGraphInitialize_ext_t           pfnAppendGraphInitialize;
+    ze_pfnAppendGraphExecute_ext_t              pfnAppendGraphExecute;
+    ze_pfnGraphGetNativeBinary_ext_t            pfnGetNativeBinary;
+    ze_pfnDeviceGetGraphProperties_ext_t        pfnDeviceGetGraphProperties;
+
+    // version 1.1
+    ze_pfnGraphGetArgumentMetadata_ext_t        pfnGraphGetArgumentMetadata;
+    ze_pfnGraphGetArgumentProperties_ext_2_t    pfnGetArgumentProperties2;
+
+    // version 1.2
+    ze_pfnGraphGetArgumentProperties_ext_3_t    pfnGetArgumentProperties3;
+
+    // version 1.3
+    ze_pfnGraphQueryNetworkCreate_ext_t             pfnQueryNetworkCreate;
+    ze_pfnGraphQueryNetworkDestroy_ext_t            pfnQueryNetworkDestroy;
+    ze_pfnGraphQueryNetworkGetSupportedLayers_ext_t pfnQueryNetworkGetSupportedLayers;
+
+    // version 1.4
+    ze_pfnGraphBuildLogGetString_ext_t          pfnBuildLogGetString;
+
+    // version 1.5
+    ze_pfnGraphCreate_ext_2_t                   pfnCreate2;
+    ze_pfnGraphQueryNetworkCreate_ext_2_t       pfnQueryNetworkCreate2;
+    ze_pfnGraphQueryContextMemory_ext_t         pfnQueryContextMemory;
+
+    // version 1.6
+    ze_pfnDeviceGetGraphProperties_ext_2_t      pfnDeviceGetGraphProperties2;
+
+    // version 1.7
+    ze_pfnGraphLoad_ext_t                       pfnLoad;
+
+} ze_graph_dditable_ext_1_7_t;
```

Comment:
I don't think this is scalable. We produce a lot of boilerplate by re-defining this struct for every API version.
Let's consider including struct size and/or version in the struct so client can call the methods depending on the API version without boilerplate code for every API release.